Create Databricks OAuth docs (#461)
SimpleDataLabsInc/prophecy-docs: docs/architecture/self-hosted/authentication/databricks-oauth.md

## OAuth use cases supported by Databricks

Databricks supports the following OAuth use cases:

- Supports Databricks acting as the Identity Provider (IdP) or 3rd party IdPs (Okta, Entra ID, Azure AD, Ping, etc.) integrated with both Databricks and Prophecy.
  - Any user within your organization can use Databricks OAuth during Pipeline development and Job configuration using their own personal identity. Individual user authentication via Databricks through an IdP can be determined by your organization.
- Works with Spark (clusters) as well as SQL (warehouses).
- Works with one or more Prophecy Fabrics.
  - Allow users to quickly establish multiple Prophecy Fabrics, each linked to a separate Databricks schema, leveraging the same OAuth connection/tokens.

## How it works

The OAuth user authentication flow includes a three-step OAuth flow to generate tokens, using Proof Key for Code Exchange (PKCE) for enhanced security.

The authentication flow then uses Prophecy-hosted callback URL to capture and process authorization codes, issuing and storing access tokens.

### Token storage

> Prophecy stores the refresh token, which are used to renew the refresh token itself, and also get a new access token to maintain authenticated connectivity to Databricks.

Reviewer: ```suggestion
Prophecy stores the refresh token, which is used to renew the refresh token itself, and also gets a new access token to maintain authenticated connectivity to Databricks.
```

<img
src={require("./img/data-bricks-oauth-select-fab.png").default}
alt="Select a Fabric"
width="70%"
/>

To proceed through the login overlays, complete the following steps:

1. Click **Continue**. A separate browser tab opens, and Databricks redirects you to the IdP registered in Databricks.

2. Log in with the IdP (or directly with Databricks if there is no IdP). The tab closes and you can proceed with activities such as Pipeline test execution.

:::note

You only need to perform this authentication periodically, depending on the OAuth timeout settings within Databricks. Your Databricks Account Administrator can adjust the timeout setting.

:::

### Project Release and Deployment

Your Prophecy Team Admin is the only member of your team able to perform deployment of the Pipelines and jobs created by the team. Prophecy uses a Databricks Service Principal via Databricks OAuth, to perform this task on behalf of the Team Admin. The Team Admin is the only user who can use this Service Principal, and only for the purpose of deploying team projects.

## OAuth Setup

The Databricks OAuth setup must be completed by both your Databricks Account Admin and your Prophecy Team Admin.

> ### Databricks Account Admin tasks

Reviewer: it looks like you need to do this step even if you aren't concerned with project release and deployment. move it up out of this section?

Your Databricks Account Admin must complete a one-time procedure on the Databricks side to register Prophecy as an OAuth App available in your account. A registration is required for each private SaaS (on-prem) customer Databricks account.

As the Databricks Account Admin, complete the following steps:

> 1. On Databricks, navigate to **Account Settings > App connections**.

Author: @ashvjit-prophecy I don't have access, so if you think it would help, could you provide screenshots of the Databricks App connections?

Reviewer 2: @alexanderahn Sure. Just let me know exactly which screenshots are needed
